fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionDebugLogging.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.http.commons.exceptionhandlers;
+
+import org.slf4j.Logger;
+
+/**
+ * @author barmintor
+ * @since 6/28/16
+ */
+public interface ExceptionDebugLogging {
+    /**
+     * Log a Throwable at the DEBUG level, log the stacktrace at the TRACE level.
+     *
+     * @param context ExceptionDebugLogging the exception intercepting context.
+     * @param error Throwable the intercepted error.
+     * @param logger Logger the logger to use
+     */
+    public default void debugException(ExceptionDebugLogging context, Throwable error, Logger logger) {
+        if (!logger.isDebugEnabled()) {
```

Comment:
If you are only making one call of `logger.debug()` and the expression used therein is simple (and I would say that an accessor is generally simple) then there is no real value in `isDebugEnabled()` compared to the extra cost. `debug()` calls it anyway, so you're going to pay for it at least once. I know you know this, @barmintor, but for anyone who doesn't, the value to using `isXEnabled()` is to avoid calculating a complex expression within a `Logger::X` call, which, because of Java semantics, happens before the logging framework executes `isXEnabled()` internally to `X()` and therefore before the logging framework can short-circuit.